### PR TITLE
rcmaehl.MSEdgeRedirect version 0.6.3.0

### PR DESCRIPTION
**Pull request by Macleykun**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/46503)

### CODE REVIEW

**Comment by Macleykun** [comment]
Can confirm locally that it works withoud any user input :) (aside of smartscreen, but that's not an issue)

File: manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Author: 'Publisher: rcmaehl'
+PackageName: MSEdgeRedirect
+PackageUrl: https://msedgeredirect.com
+License: Robert Maehl, using LGPL 3 License
+LicenseUrl: https://github.com/rcmaehl/MSEdgeRedirect/blob/main/LICENSE
+ShortDescription: A Tool to Redirect News, Search, Widgets, Weather and More to Your Default Browser
+Moniker: MSEdgeRedirect
+Tags:
+- Redirect
+ReleaseNotes: 0.6.3.0
```
Comment (KaranKad) [inline manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml:19]:
```suggestion
Moniker: msedgeredirect
Tags:
- redirect
ReleaseNotes: |
 * Fixed Edge Profile issues
 * Fixed Settings Start Menu Item
 * Fixed old as heck misspelling in Installer
 * Fixed Start Search Suggestions not working
 * Minor Redesign to Mode and Settings Menu
 * Fixed --from-installer tag being redirected
 * Inactive code for de-embedding news articles
 * Fixed Silent/Deployed Installs missing Start Menu entries
 * Added Cleanup Script to GitHub Assets folder for botched installs
 * Fixed HTTP/HTTPS wanting an app to be selected despite no changes 
```

File: manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 0.5.0.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.1.0.schema.json
+
+PackageIdentifier: rcmaehl.MSEdgeRedirect
+PackageVersion: 0.6.3.0
+PackageLocale: en-US
+Publisher: rcmaehl
+PublisherUrl: https://msedgeredirect.com
+PublisherSupportUrl: https://github.com/rcmaehl/MSEdgeRedirect/issues
+Author: 'Publisher: rcmaehl'
```
Comment (KaranKad) [inline manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml:10]:
```suggestion
Author: Robert C. Maehl
```

File: manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+PackageIdentifier: rcmaehl.MSEdgeRedirect
+PackageVersion: 0.6.3.0
+PackageLocale: en-US
+Publisher: rcmaehl
+PublisherUrl: https://msedgeredirect.com
+PublisherSupportUrl: https://github.com/rcmaehl/MSEdgeRedirect/issues
+Author: 'Publisher: rcmaehl'
+PackageName: MSEdgeRedirect
+PackageUrl: https://msedgeredirect.com
+License: Robert Maehl, using LGPL 3 License
```
Comment (KaranKad) [inline manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml:13]:
```suggestion
License: LGPL-3.0
```

**Review by KaranKad — CHANGES_REQUESTED** [review]

**Comment by KaranKad** [comment]
x86 version is also available. Might as well add it.
https://github.com/rcmaehl/MSEdgeRedirect/releases/download/0.6.3.0/MSEdgeRedirect_x86.exe

File: manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml (hunk 1 above)
Comment (Macleykun) [reply manifests/r/rcmaehl/MSEdgeRedirect/0.6.3.0/rcmaehl.MSEdgeRedirect.locale.en-US.yaml:19]: Yeah that makes alot of sense, should've known to use | followed by the changelog itself obviously

**Comment by Trenly** [comment]
#42616

**Comment by Macleykun** [comment]
> x86 version is also available. Might as well add it. https://github.com/rcmaehl/MSEdgeRedirect/releases/download/0.6.3.0/MSEdgeRedirect_x86.exe

Mm, didn't knew i could add multiple, will do!

**Comment by Macleykun** [comment]
> > x86 version is also available. Might as well add it. https://github.com/rcmaehl/MSEdgeRedirect/releases/download/0.6.3.0/MSEdgeRedirect_x86.exe
> 
> Mm, didn't knew i could add multiple, will do!

How do i actually add the 32-bit? seperate pull?
Probably need to make some changes at our end aswell to auto update the 32-bit aswell

**Comment by KaranKad** [comment]
@Macleykun 
Take the manifest below as an example for multiple architectures. You can add this by editing the file from github itself.
https://github.com/microsoft/winget-pkgs/pull/44323/files

**Comment by Macleykun** [comment]
> @Macleykun Take the manifest below as an example for multiple architectures. You can add this by editing the file from github itself. https://github.com/microsoft/winget-pkgs/pull/44323/files

done! thanks again for checking and suggesting! <3

**Comment by Macleykun** [comment]
> Hello @Macleykun,
> The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

Not exactly sure what went wrong, I installed this locally and everything went perfect, the logs from the CI don’t specify what went wrong. Is this something I can fix or is this a known issue with the CI?

**Comment by KaranKad** [comment]
It fails to install inside sandbox with exit code: 4315
image

**Comment by Macleykun** [comment]
@rcmaehl any guess why this error is triggered?
https://github.com/rcmaehl/MSEdgeRedirect/blob/4254eefd1c1651e1a5a7f61023874dd034cac8de/MSEdgeRedirect.au3#L356

I’m gonna assume this is because of the sandbox environment?

**Comment by rcmaehl** [comment]
> @rcmaehl any guess why this error is triggered? https://github.com/rcmaehl/MSEdgeRedirect/blob/4254eefd1c1651e1a5a7f61023874dd034cac8de/MSEdgeRedirect.au3#L356
> 
> I’m gonna assume this is because of the sandbox environment? Actually pretty sure because of it not having edge in that environment…

It's a failsafe to make sure the program doesn't get stuck in an infinite loop. I'll see if I can adjust that.

**Comment by rcmaehl** [comment]
I've just adjusted the check as well as the architecture check for winget installs. Can we verify that Winget won't let users install 32-bit versions on 64-bit machines if a 64-bit version is available?

**Comment by Macleykun** [comment]
> I've just adjusted the check as well as the architecture check for winget installs. Can we verify that Winget won't let users install 32-bit versions on 64-bit machines if a 64-bit version is available?

I believe that winget will get the appropriate version for the OS (unless the user specifies it explicit).
Also, can you make a new release so I can edit the manifests so we can get them green checks :) ? (Don’t wanna make a new pull again haha)

**Comment by Macleykun** [comment]
> Hello @Macleykun, The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

We have fixed the issue but are running into some issues for the end user. We'll make a new release once we got that ironed out aswell!
Then i'll update the manifest download and hash to use the newer version instead of making another pull.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
> We have fixed the issue but are running into some issues for the end user. We'll make a new release once we got that ironed out aswell!
> Then i'll update the manifest download and hash to use the newer version instead of making another pull.

You can set this PR to DRAFT mode until it's fixed.

**Comment by denelon** [comment]
@Macleykun will the package install unattended if Edge isn't present?

**Comment by rcmaehl** [comment]
> @Macleykun will the package install unattended if Edge isn't present?

Yes

**Comment by Macleykun** [comment]
> Hello @Macleykun, The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

what went wrong? (sorry i still can't find any logs that say what the error is why it went wrong)

**Comment by SpecterShell** [comment]
https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip
This log may help you. It seems that the installer still returns `4315`. 🤔

**Comment by Macleykun** [comment]
> https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip This log may help you. It seems that the installer still returns `4315`. 🤔

i do thank you alot for this link, however i was looking into the .txt files and saw this error?
Installation failed with exit code -1978335226

@gnpaone you know why this happend?

**Comment by gnpaone** [comment]
> > https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip This log may help you. It seems that the installer still returns `4315`. 🤔
> 
> i do thank you alot for this link, however i was looking into the .txt files and saw this error?
> Installation failed with exit code -1978335226
> 
> @gnpaone you know why this happend?

I'll look at it

**Comment by gnpaone** [comment]
> https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip This log may help you. It seems that the installer still returns `4315`. 🤔

May I please know what `4315` error code refers to? I have seen error code `87` previously elsewhere and it means error in manifests, is there similar meaning to this?

**Comment by rcmaehl** [comment]
> > https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip This log may help you. It seems that the installer still returns `4315`. 🤔
> 
> May I please know what `4315` error code refers to? I have seen error code `87` previously elsewhere and it means error in manifests, is there similar meaning to this?

Error codes for the package are here:

https://github.com/rcmaehl/MSEdgeRedirect/wiki/Deploying-MSEdgeRedirect#return-values

**Comment by gnpaone** [comment]
> > > https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/278914/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip This log may help you. It seems that the installer still returns `4315`. 🤔
> > 
> > 
> > May I please know what `4315` error code refers to? I have seen error code `87` previously elsewhere and it means error in manifests, is there similar meaning to this?
> 
> Error codes for the package are here:
> 
> https://github.com/rcmaehl/MSEdgeRedirect/wiki/Deploying-MSEdgeRedirect#return-values
> 
> 4315 should have been fixed in the recent test (CI) builds, and I can release 0.6.3.1 if needed.

Yah, I understood the reason mentioned for 4315. I think yes for releasing 0.6.3.1 if 4315 error won't occur

**Comment by ItzLevvie** [comment]
image
image

**Comment by gnpaone** [comment]
> image image

Thanks for the pics, will try to resolve the issue

**Comment by Macleykun** [comment]
We’re still working on it.

**Pull request closed without merging**